src/main/scala/scalaz/nio/ByteBuffer.scala

```diff
@@ -0,0 +1,61 @@
+package scalaz.nio
+
+import scalaz.zio.IO
+
+import java.nio.{ ByteOrder, ByteBuffer => JByteBuffer }
+
+class ByteBuffer private (private[nio] val javaBuffer: JByteBuffer)
+    extends Buffer[Byte, JByteBuffer](javaBuffer) {
+
+  type Self = ByteBuffer
+
+  def array: IO[Exception, Array[Byte]] = IO.syncException(javaBuffer.array())
+
+  def order: IO[Nothing, ByteOrder] = IO.now(javaBuffer.order())
+
+  def slice: IO[Exception, ByteBuffer] = IO.syncException(javaBuffer.slice()).map(new ByteBuffer(_))
+
```

Comment:
`ByteBuffer` has extra this all crazy methods `getChar: Char`. Are they somewhere here? =)
https://github.com/scalaz/scalaz-nio/blob/eaa22aaa059a4a5a0780e73d4c85584304ce638b/src/main/scala/scalaz/nio/Buffer.scala#L82-L86

Reply:
No, I will add them in next commit.